frontend/public/components/namespace-dropdown.tsx

```diff
@@ -0,0 +1,309 @@
+import * as React from 'react';
+import {
+  Button,
+  Divider,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStatePrimary,
+  Menu,
+  MenuContent,
+  MenuGroup,
+  MenuInput,
+  MenuItem,
+  MenuList,
+  Switch,
+  TextInput,
+  Title,
+} from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { useUserSettingsCompatibility } from '@console/shared';
+import { isSystemNamespace } from './factory/table-filters';
+import NamespaceMenuToggle from './namespace-menu-toggle';
+
+const NamespaceBarDropdown_ = (props: NamespaceBarDropdownProps) => {
```

Comment:
I have tried to stay way from typing functions using React.FC because it will allow untyped children.  This particular component (`NamespaceBarDropdown`) should not have any children.  By typing it using React.FC, then children can be passed into this component without a typescript error.

With that said, consistency is a good thing.   Changes in next commit.

Reply:
Yeah, I'm actually okay either way. We've tended to take a gradual approach to other tech-debt issues, and we could do the same here. I think it would be good to start moving away from React.FC. We'll just need to start enforcing that consistently.